File: CLOMI/workshop/ressources/annexe/python/Kafka/consumer.py
from kafka import KafkaConsumer
import json

def KafkaReadMessage(topic: str, group_id_: str, routine: function):
    consumer = KafkaConsumer(
        topic,
        bootstraps_servers='localhost:9092',
        group_id=group_id_,
        value_deserializer=lambda m: json.loads(m.decode('utf-8'))
    )
    for msg in consumer:
        routine(msg)

File: CLOMI/workshop/ressources/annexe/node/Kafka/consumer.js
import { Kafka } from 'kafkajs'

const kafka = new Kafka({
    clientId: "app",
    brokers: ['localhost:9092', 'localhost:9092']
})

const consumer = kafka.consumer({ groupId: 'kafka' });

await consumer.connect();

export const KafkaReadMessage = async (topic, routine) => {
    await consumer.subscribe({ topic: topic, fromBeginning: true });

    await consumer.run({
        eachMessage: async ({ _, _, message }) => {
            routine(message)
        },
    });
}
